### DISCUSSION TITLE
dvc.lock - file size changes while md5 hash remains the same? (between OSX and Windows)

### DISCUSSION THREAD

**Discussion by SebBanDev** [treeverse/dvc#6313] [Help]
I have some issues using a s3 bucket to push and pull input data with dvc.

I have a stage where the `deps` are `outs` of a prior stage.

When I reproduce the dvc.yaml and push the data on OSX(Mac)
and then pull it on another machine using Windows, I reproduce the dvc.yaml again,
dvc.lock shows the same hash but different file sizes for that stage `deps` and reruns the stage completely, then also showing different files sizes and hash on the `outs`

git marks the diff in my IDE in the deps of that stage in the .lock file only for sizes:

![image](https://user-images.githubusercontent.com/51693106/125586142-de63d8f3-aaf2-4c11-a6df-e45a7a4d34db.png)

Any idea how the file size in `deps` could change if the hash is the same?

What I tried so far:
deleting the files on OSX, pull them again from s3, reproducing dvc.yaml -> no changes detected, dvc.lock stays the same
delteing the files on Windows, pull them again from s3, reproducing dvc.yaml -> changes detected, dvc.lock shows different file sizes for the files in deps of that stage

On both systems, I definitely use the same git commit.

**Comment by pmrowla** [comment]
Does your pipeline generate the csv's with different line endings depending on OS (i.e. `\n` on OSX, and `\r\n` on windows)?

DVC computes hashes by doing a dos2unix conversion for text files, but iirc the file sizes written to dvc.lock are read directly from the filesystem, which could lead to getting consistent hashes but different sizes in a case like this (because the hash generated on windows will be for the file without carriage returns, but the size will be for the file with carriage returns).

**Reply by SebBanDev** [reply]
I had the same suspicion, I will try to setup my IDE so that it always uses unix line endings and check which line endings are used in both systems running dvc repro

**Reply by SebBanDev** [reply]
Can I configure dvc to enforce the same line endings of files as in s3 bucket (unix)? In my Windows machine the IDE is already set to using unix as default. nevertheless the .csvs seem to have Windows line endings if I delete and pull them.

**Reply by SebBanDev** [reply]
Or is there a way to turn of the file size check as long as the hash is the same?
